controllers/storage/provision.go

```diff
@@ -250,64 +298,76 @@ func (m *MinIOReconciler) generateOssSecret() *corev1.Secret {
 		},
 		Type: corev1.SecretTypeOpaque,
 		Data: map[string][]byte{
-			"authurl":             []byte(m.HarborCluster.Spec.Storage.Swift.Authurl),
-			"username":            []byte(m.HarborCluster.Spec.Storage.Swift.Username),
-			"password":            []byte(m.HarborCluster.Spec.Storage.Swift.Password),
-			"container":           []byte(m.HarborCluster.Spec.Storage.Swift.Container),
-			"region":              []byte(m.HarborCluster.Spec.Storage.Swift.Region),
-			"tenant":              []byte(m.HarborCluster.Spec.Storage.Swift.Tenant),
-			"tenantid":            []byte(m.HarborCluster.Spec.Storage.Swift.Tenantid),
-			"domain":              []byte(m.HarborCluster.Spec.Storage.Swift.Domain),
-			"Domainid":            []byte(m.HarborCluster.Spec.Storage.Swift.Domainid),
-			"trustid":             []byte(m.HarborCluster.Spec.Storage.Swift.Trustid),
-			"insecureskipverify":  []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.Swift.Insecureskipverify)),
-			"prefix":              []byte(m.HarborCluster.Spec.Storage.Swift.Prefix),
-			"secretkey":           []byte(m.HarborCluster.Spec.Storage.Swift.Secretkey),
-			"authversion":         []byte(string(m.HarborCluster.Spec.Storage.Swift.AuthVersion)),
-			"endpointtype":        []byte(m.HarborCluster.Spec.Storage.Swift.EndpointType),
-			"tempurlcontainerkey": []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.Swift.TempurlContainerkey)),
-			"tempurlmethods":      []byte(m.HarborCluster.Spec.Storage.Swift.TempurlMethods),
-			"chunksize":           []byte(m.HarborCluster.Spec.Storage.Swift.Chunksize)},
+			"accesskeyid":     []byte(m.HarborCluster.Spec.Storage.Oss.Accesskeyid),
+			"accesskeysecret": []byte(m.HarborCluster.Spec.Storage.Oss.Accesskeysecret),
+			"region":          []byte(m.HarborCluster.Spec.Storage.Oss.Region),
+			"bucket":          []byte(m.HarborCluster.Spec.Storage.Oss.Bucket),
+			"endpoint":        []byte(m.HarborCluster.Spec.Storage.Oss.Region),
+			"internal":        []byte(m.HarborCluster.Spec.Storage.Oss.Internal),
+			"encrypt":         []byte(m.HarborCluster.Spec.Storage.Oss.Encrypt),
+			"secure":          []byte(m.HarborCluster.Spec.Storage.Oss.Secure),
+			"chunksize":       []byte(m.HarborCluster.Spec.Storage.Oss.Secure),
+			"rootdirectory":   []byte(m.HarborCluster.Spec.Storage.Oss.RootDirectory),
+		},
 	}
 }
 
 func (m *MinIOReconciler) Provision() (*lcm.CRStatus, error) {
 	// TODO remove mcs secret ref https://github.com/minio/minio-operator/blob/master/examples/minioinstance.yaml
 	//mcsSecret := m.generateMcsSecret()
-	//err := m.KubeClient.Create(m.Ctx, mcsSecret)
+	//err := m.KubeClient.Create(mcsSecret)
 	//if err != nil {
 	//	return minioNotReadyStatus(ErrorReason2, err.Error()), err
 	//}
 	credsSecret := m.generateCredsSecret()
-	err := m.KubeClient.Create(m.Ctx, credsSecret)
+	err := m.KubeClient.Create(credsSecret)
 	if err != nil {
 		return minioNotReadyStatus(ErrorReason2, err.Error()), err
 	}
 	service := m.generateService()
```

Comment:
For minIO, its CR is not a full accessible service? Why do you need to create service for minIO?

Reply:
yes, we must add an external service to enable Service-based access to the MinIOInstance pods.